client/my-sites/posts/post-list.jsx

```diff
@@ -261,7 +261,7 @@ var Posts = React.createClass( {
 			i;
 
 		// posts have loaded, sites have loaded, and we have a site instance or are viewing all-sites
-		if ( posts.length && this.props.sites.initialized ) {
+		if ( posts.length && this.props.haveSites ) {
```

Comment:
`initialized` doesn't entail fresh data, only localStorage-bootstrapped data. In a Redux world, where we load persisted state before the initial render call, I don't know how removing this altogether would impact the logic.